### ISSUE TITLE
assertion failed in sublime/default.nix

### ISSUE THREAD

**Issue by bryaan** [NixOS/nixpkgs#34832]
## Issue description

Trying to install sublime on macos (High Sierra)

```bash
 $ nix-env -iA nixpkgs.funopsEnv --show-trace
installing ‘funopsEnv’
error: while evaluating the attribute ‘pkgs’ of the derivation ‘funopsEnv’ at /nix/store/k8mvnyc9fs9cywsxp0jpnk6wrgcf8v17-nixpkgs-18.03pre125635.846b5c42f19/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:148:11:
while evaluating ‘callPackageWith’ at /nix/store/k8mvnyc9fs9cywsxp0jpnk6wrgcf8v17-nixpkgs-18.03pre125635.846b5c42f19/nixpkgs/lib/customisation.nix:113:35, called from /nix/store/k8mvnyc9fs9cywsxp0jpnk6wrgcf8v17-nixpkgs-18.03pre125635.846b5c42f19/nixpkgs/pkgs/top-level/all-packages.nix:17133:13:
while evaluating ‘makeOverridable’ at /nix/store/k8mvnyc9fs9cywsxp0jpnk6wrgcf8v17-nixpkgs-18.03pre125635.846b5c42f19/nixpkgs/lib/customisation.nix:72:24, called from /nix/store/k8mvnyc9fs9cywsxp0jpnk6wrgcf8v17-nixpkgs-18.03pre125635.846b5c42f19/nixpkgs/lib/customisation.nix:117:8:
while evaluating anonymous function at /nix/store/k8mvnyc9fs9cywsxp0jpnk6wrgcf8v17-nixpkgs-18.03pre125635.846b5c42f19/nixpkgs/pkgs/applications/editors/sublime/default.nix:1:1, called from /nix/store/k8mvnyc9fs9cywsxp0jpnk6wrgcf8v17-nixpkgs-18.03pre125635.846b5c42f19/nixpkgs/lib/customisation.nix:74:12:
assertion failed at /nix/store/k8mvnyc9fs9cywsxp0jpnk6wrgcf8v17-nixpkgs-18.03pre125635.846b5c42f19/nixpkgs/pkgs/applications/editors/sublime/default.nix:5:1
```

Same thing a little easier to read:

```bash
 $ nix-env -iA nixpkgs.funopsEnv --show-trace
installing ‘funopsEnv’
error: while evaluating the attribute ‘pkgs’ of the derivation ‘funopsEnv’ at .../nixpkgs/pkgs/stdenv/generic/make-derivation.nix:148:11:
while evaluating ‘callPackageWith’ at .../nixpkgs/lib/customisation.nix:113:35, called from .../nixpkgs/pkgs/top-level/all-packages.nix:17133:13:
while evaluating ‘makeOverridable’ at .../nixpkgs/lib/customisation.nix:72:24, called from  .../nixpkgs/lib/customisation.nix:117:8:
while evaluating anonymous function at .../pkgs/applications/editors/sublime/default.nix:1:1, called from .../nixpkgs/lib/customisation.nix:74:12:
assertion failed at  .../nixpkgs/pkgs/applications/editors/sublime/default.nix:5:1
```

### Steps to reproduce

```nix
# ~/.nixpkgs/config.nix
{
  allowUnfree = true;
  packageOverrides = pkgs: with pkgs; {
    funopsEnv = with pkgs; buildEnv {
      name = "funopsEnv";
      paths = [
        sublime
      ];
    };
  };
}
```

## Technical details

 - system: `"x86_64-darwin"`
 - host os: `Darwin 17.4.0, macOS 10.13.3`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `"nixpkgs-18.03pre125635.846b5c42f19"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

**Comment by rnhmjoj** [comment]
The assert failing is that your system is not linux: the package only handle the linux binaries currently.
I'm not sure why this is implemented using an assert instead of setting `meta.platforms = platforms.linux`, that would give you a comprehensible error.
Anyway you could try to edit the package but I'd say the depedencies for darwin are quite different from linux so it could be complicated.

**Comment by matthewbauer** [comment]
Fixed in 143978a477e4891bd94be6d9fcf257f726b403cd
